### DISCUSSION TITLE
[rfc] Simplify names for key patterns and use Frictionless prefix for them

### DISCUSSION THREAD

**Discussion by rufuspollock** [frictionlessdata/frictionlessdata.io#639] [Ideas] [answered]
This is just an idea and i'm not sure about it so thoughts are welcome:

* Data Package => [Frictionless] Dataset (or [Frictionless] Package)
* Data Resource => [Frictionless] Resource
* Table Schema => [Frictionless] Schema

Why?

* Get down to one word consistently (and drop redundant "Data")
* Use Frictionless consistently as a way to distinguish and brand the patterns

/cc @pwalsh

**Comment by cpina** [comment]
My 2 cents: I like "Table Schema" to know that this is the schema of a table... and in case that some day there were other schemas (I don't know, "Document schema" or something else).

For the other too I'm happy either way, maybe removing "Data" as suggested helps to read it (fewer redundant words, etc.).

**Comment by pwalsh** [comment]
> Data Package => [Frictionless] Dataset (or [Frictionless] Package)
> Data Resource => [Frictionless] Resource
> Table Schema => [Frictionless] Schema

Adding Frictionless at the front is fine, for branding purposes. But, I definitely would not get rid of "Data" or "Table" - the names are perfectly clear at present.

**Comment by rufuspollock** [comment]
@pwalsh adding Frictionless at front does make them long and if you had `Dataset` rather than Package you would not need `Data`. I agree for `Resource` it might be needed though i have wondered if `File` would be better.

hear you both on keeping Table

**Comment by cpina** [comment]
Thinking aloud: to me `File` is a type of `Resource`... but there could be other type of resources besides files. E.g. if we had support for a object storage alike technically it would be an `Object`.

If we renamed it to `File` it would be limited to files only... no objects or other resources.

But `File` is clearer for newcomers I think (more explicit)

**Comment by lwinfree** [comment]
- I think Frictionless Package is fine (although I don't mind `data` package)
- I think Carles has a good point about `File` vs `Resource`, but I have seen some users being confused about the term `Resource`. A fix here could be as simple as making sure we describe this term more fully in our docs
- Agree that keeping `Table` is important in the schema name

@roll do you have thoughts here too?

**Comment by roll** [comment]
In the software, we e.g. have `Package`, `Resource`, `Schema` so in the documentation it can be prefixed by any of `Frictionless` or `Data/Table`. Honestly, I don't know what's better branding vs using more general concepts. At the same time, I don't see a problem if e.g. it's Frictionless Schema or Frictionless Package - we will get used to it very soon if it's used consistently. 

Just a few examples - there is a Pandas Dataframe not just a data frame, a Postgres table not just a table, a Jupiter Notebook not just a notebook.

In the case of Frictionless Package, will it mean that Frictionless Package is an implementation of the more general concept of Data Package?

**Comment by rufuspollock** [comment]
Chatting with @pwalsh sense that `Dataset` is clearer than `Package`. Everything else can stay as is though perhaps we can start dropping `Data` in front of resource by default.

So, what I'm inclining towards recommending is:

```
Data Package  =>  Dataset
Data Resource => [Data] Resource
Table Schema  =>  Table Schema
```

**Comment by rufuspollock** [comment, marked as answer]
AGREED:

```
Data Package  => [Frictionless] Dataset
Data Resource => [Frictionless] Resource
Table Schema  => [Frictionless] Table Schema
```
